pkg/validator/validator.go

```diff
@@ -49,12 +48,24 @@ func New(client openapi.Client) (*Validator, error) {
 // It will return errors when there is an issue parsing the object, or if
 // it contains fields unknown to the schema, or if the schema was recursive.
 func (s *Validator) Parse(document []byte) (schema.GroupVersionKind, *unstructured.Unstructured, error) {
-	metadata := metav1.TypeMeta{}
-	if err := yaml.Unmarshal(document, &metadata); err != nil {
+	// Initialize an empty unstructured object
+	obj := &unstructured.Unstructured{}
+
+	// Parse the entire document into the unstructured object
+	if err := yaml.Unmarshal(document, obj); err != nil {
 		return schema.GroupVersionKind{}, nil, fmt.Errorf("failed to parse yaml: %w", err)
 	}
 
-	gvk := metadata.GetObjectKind().GroupVersionKind()
+	// get the object name
+	objName := obj.GetName()
+	if len(objName) == 0 {
+		return schema.GroupVersionKind{}, nil, fmt.Errorf("object name cannot be empty")
+	}
```

Comment:
I think this is validated elsewhere isn't it. I think it should be removed. It is OK for the new error to be missing the name

Reply:
hmm I can take a look again, but sure no problem Ill remove it